python/cudf/cudf/tests/test_orc.py

```diff
@@ -1926,3 +1926,28 @@ def test_orc_chunked_writer_stripe_size(datadir):
 
     orc_file = orc.ORCFile(buffer)
     assert_eq(orc_file.nstripes, 5)
+
+
+def test_reader_lz4():
+    from pyarrow import orc
+
+    pdf = pd.DataFrame({"ints": [1, 2] * 5001})
+    pa_table = pa.Table.from_pandas(pdf)
+
+    buffer = BytesIO()
+    writer = orc.ORCWriter(buffer, compression="LZ4")
+    writer.write(pa_table)
+    writer.close()
+
+    got = cudf.read_orc(buffer)
+    assert_eq(pdf, got)
+
+
+def test_writer_lz4():
+    gdf = cudf.DataFrame({"ints": [1, 2] * 5001})
+
+    buffer = BytesIO()
+    gdf.to_orc(buffer, compression="lz4")
```

Comment:
I think @bdice had a similar question somewhere:  Is it `compression="LZ4"` or `compression="lz4"`. Or does the case not matter?

Reply:
The case matters, unfortunately. I used lz4 (lowercase) to match Pandas, since we're supposed to be a drop-in replacement where possible. Pandas seemingly does not support LZ4 in to_orc, and pyarrow uses uppercase names. I used lowercase in ORC as well to keep consistent with cudf.to_parquet(which is trying to be consistent with pd.to_parquet :D).
I'm happy to take suggestions here.
CC cuIO Python API guru @galipremsagar

Reply:
responded here: https://github.com/rapidsai/cudf/pull/14906#discussion_r1486000151